fastscanner/services/scanners/service.py

```diff
@@ -67,30 +68,48 @@ async def handle(self, channel_id: str, data: dict[Any, Any]) -> None:
 
 
 class ScannerService:
-    def __init__(self, candles: CandleStore, channel: Channel):
+    def __init__(
+        self, candles: CandleStore, channel: Channel, symbols_provider: SymbolsProvider
+    ):
         self._candles = candles
         self._channel = channel
-        # self._handlers: dict[str, ScannerChannelHandler] = {}  # Remove if not needed
+        self._symbols_provider = symbols_provider
+        self._handlers: dict[str, list[ScannerChannelHandler]] = {}
 
     async def subscribe_realtime(
         self,
-        symbol: str,
-        freq: str,
-        scanner: Scanner,
+        params: ScannerParams,
         handler: SubscriptionHandler,
-    ):
-        stream_key = f"candles_min_{symbol}"
+        freq: str,
+    ) -> str:
+        scanner = ScannersLibrary.instance().get(params.type_, params.params)
         scanner_id = scanner.id()
-        handler_id = f"{scanner_id}_{symbol}"
-        sch = ScannerChannelHandler(symbol, scanner, handler, freq)
-        # self._handlers[handler_id] = sch  # Remove if not needed
-        await self._channel.subscribe(stream_key, sch)
 
-    async def unsubscribe_realtime(
-        self,
-        channel_id: str,
-        scanner_id: str,
-        symbol: str,
-    ):
-        handler_id = f"{scanner_id}_{symbol}"
-        await self._channel.unsubscribe(channel_id, handler_id)
+        symbols = await self._symbols_provider.active_symbols()
+        symbols = symbols[:1000]
+
+        async def subscribe_symbol(symbol: str) -> ScannerChannelHandler:
```

Comment:
this can be a private class method and pass self._subscribe_symbol to asyncio.create_task